### PR TITLE
OpenTelemetry Exporter typedoc and exporter options update

### PR DESCRIPTION
**Pull request by hectorhdzg**
Adding exporter options as explicit parameter
Fixing issues with typedoc return types

### CODE REVIEW

File: sdk/monitor/microsoft-opentelemetry-exporter-azuremonitor/microsoft/opentelemetry/exporter/azuremonitor/export/_base.py [hunk 1]
```diff
@@ -36,16 +36,10 @@ class BaseExporter:
...
-        :param options: Exporter configuration options.
-        :type options: ~microsoft.opentelemetry.exporter.azuremonitor.options.ExporterOptions
-        """
-        options = ExporterOptions(**kwargs)
+    def __init__(self, options: ExporterOptions = None, **kwargs):
+        # type: (...) -> None
         parsed_connection_string = ConnectionStringParser(
-            options.connection_string)
-
+            options.connection_string if options is not None else None)
```
Comment (lzchen) [inline sdk/monitor/microsoft-opentelemetry-exporter-azuremonitor/microsoft/opentelemetry/exporter/azuremonitor/export/_base.py:42]: I think we want to fast fail if connection string is empty or not supplied.

File: sdk/monitor/microsoft-opentelemetry-exporter-azuremonitor/microsoft/opentelemetry/exporter/azuremonitor/export/_base.py [hunk 2]
```diff
@@ -36,16 +36,10 @@ class BaseExporter:
...
     """
 
-    def __init__(self, **kwargs):
-        """Azure Monitor base exporter for OpenTelemetry.
-
-        :param options: Exporter configuration options.
-        :type options: ~microsoft.opentelemetry.exporter.azuremonitor.options.ExporterOptions
-        """
-        options = ExporterOptions(**kwargs)
+    def __init__(self, options: ExporterOptions = None, **kwargs):
```
Comment (lzchen) [inline sdk/monitor/microsoft-opentelemetry-exporter-azuremonitor/microsoft/opentelemetry/exporter/azuremonitor/export/_base.py:39]: We want users to be able to do:
1. AzureSpanExporter(connection_string=...)
2. AzureSpanExporter() <-- connection string/ikey from env var
3. AzureSpanExporter(connection_string=..., <other-kwargs-used-by-azure-core-classes)
4. AzureSpanExporter(<other-kwargs-used-by-azure-core-classes) <-- connection string/ikey from env var

Are all these possible with this signature?

**Review by lmazuel — CHANGES_REQUESTED** [review]
Do NOT use type comment annotations. Your code is Python 3 only, comment type annotations are here solely for Python 2.7 compatibility.

**Comment by hectorhdzg** [comment]
Closing this as is apparently not needed, added Python 2 style comment annotations to have better results while running apistub

**Pull request closed without merging**